src/player.c

```diff
@@ -530,7 +530,12 @@ static inline void read_row(struct context_data *ctx, int pat, int row)
 		}
 
 		if (check_delay(ctx, &ev, chn) == 0) {
-			if (!f->rowdelay_set || f->rowdelay > 0) {
+			/* rowdelay_set bit 1 is set only in the first tick of the row
+			 * event if the delay causes the tick count resets to 0. We test
+			 * it to read row events only in the start of the row. (see the
+			 * OpenMPT test case FineVolColSlide.it)
+			 */
+			if (!f->rowdelay_set || ((f->rowdelay_set & 2) && f->rowdelay > 0)) {
 				libxmp_read_event(ctx, &ev, chn);
```

Comment:
Since `f->rowdelay_set` and `f->rowdelay` are 0 at this point on the first tick of a row it might be possible to omit the ` || ((f->rowdelay_set & 2) && f->rowdelay > 0)` part altogether. I haven't verified this and the new version works as-intended so it is probably not worth changing.

Reply:
There's plenty of room for refactoring and clean-ups there. The role of bit 1 in rowdelay_set perhaps could be better represented by a true tick counter (which doesn't wrap at the current speed).